crates/db_schema/src/impls/local_user.rs

```diff
@@ -49,18 +50,39 @@ impl LocalUser {
     pool: &mut DbPool<'_>,
   ) -> Result<Vec<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
-    diesel::update(local_user)
-      .set(accepted_application.eq(true))
+    diesel::update(local_user::table)
+      .set(local_user::accepted_application.eq(true))
       .get_results::<Self>(conn)
       .await
   }
 
-  pub async fn is_email_taken(pool: &mut DbPool<'_>, email_: &str) -> Result<bool, Error> {
+  pub async fn delete_old_denied_local_users(pool: &mut DbPool<'_>) -> Result<usize, Error> {
+    let conn = &mut get_conn(pool).await?;
+
+    // Make sure:
+    // - The deny reason exists
+    // - The app is older than a week
+    // - The accepted_application is false
+    let old_denied_registrations = registration_application::table
+      .filter(registration_application::deny_reason.is_not_null())
+      .filter(registration_application::published.lt(now() - 1.week()))
+      .select(registration_application::local_user_id);
+
+    // Delete based on join logic is here:
+    // https://stackoverflow.com/questions/60836040/how-do-i-perform-a-delete-with-sub-query-in-diesel-against-a-postgres-database
+    let users = local_user::table
+      .filter(local_user::id.eq_any(old_denied_registrations))
+      .filter(local_user::accepted_application.eq(false));
+
+    diesel::delete(users).execute(conn).await
```

Comment:
I think you also need to explicitly delete from person table.

Reply:
Deleting the person should automaticaly delete the local user, so this could be done to have only 1 statement:

```
let users = person::table
    .filter(exists(local_user::table
        .filter(local_user::id.eq_any(old_denied_registrations))
        .filter(not(local_user::accepted_application))
    ));
```

Reply:
Oops good call. I'll do that now.

Reply:
Pushed up changes. I don't think you can use `exists(...` on another when it doesn't know what columns to check, so I used `person::id.eq_any`